source/opt/inline_pass.cpp

```diff
@@ -0,0 +1,430 @@
+  ir::Function* calleeFn = id2function_[calleeId];
+
+  // Map parameters to actual arguments
+  int i = 0;
+  calleeFn->ForEachParam(
+      [&call_ii, &i, &callee2caller](const ir::Instruction* cpi) {
+        const uint32_t pid =
+            cpi->GetOperand(kSpvFunctionParameterResultId).words[0];
+        callee2caller[pid] =
+            call_ii->GetOperand(kSpvFuncitonCallArgumentId + i).words[0];
+        i++;
+      });
+
+  // Define caller local variables for all callee variables and create map to
+  // them
+  auto cbi = calleeFn->begin();
+  auto cvi = cbi->begin();
+  while (cvi->opcode() == SpvOp::SpvOpVariable) {
+    std::unique_ptr<ir::Instruction> var_inst(new ir::Instruction(*cvi));
+    uint32_t newId = TakeNextId();
+    var_inst->SetResultId(newId);
+    callee2caller[cvi->result_id()] = newId;
+    newVars->push_back(std::move(var_inst));
+    cvi++;
+  }
+
+  // Create return var if needed
+  uint32_t returnVarId = 0;
+  const uint32_t calleeTypeId = calleeFn->GetTypeId();
+  const ir::Instruction* calleeType =
+      def_use_mgr_->id_to_defs().find(calleeTypeId)->second;
+  if (calleeType->opcode() != SpvOpTypeVoid) {
+    // find or create ptr to callee return type
+    uint32_t returnVarTypeId = FindPointerToType(calleeTypeId);
+    if (returnVarTypeId == 0) returnVarTypeId = AddPointerToType(calleeTypeId);
+    // Add return var to new function scope variables
+    returnVarId = TakeNextId();
+    std::vector<ir::Operand> in_operands;
+    in_operands.emplace_back(
+        spv_operand_type_t::SPV_OPERAND_TYPE_STORAGE_CLASS,
+        std::initializer_list<uint32_t>{uint32_t(SpvStorageClassFunction)});
+    std::unique_ptr<ir::Instruction> var_inst(new ir::Instruction(
+        SpvOpVariable, returnVarTypeId, returnVarId, in_operands));
+    newVars->push_back(std::move(var_inst));
+  }
+
+  // Clone and map callee code
+  bool prevInstWasReturn = false;
+  uint32_t returnLabelId = 0;
+  bool multiBlocks = false;
+  std::unique_ptr<ir::BasicBlock> bp;
+  calleeFn->ForEachInst(
+      [&newBlocks, &callee2caller, &call_bi, &call_ii, &bp, &prevInstWasReturn,
+       &returnLabelId, &returnVarId, &calleeTypeId, &multiBlocks, &postCallSI,
+       &preCallSI, this](const ir::Instruction* cpi) {
+        switch (cpi->opcode()) {
+          case SpvOpFunction:
+          case SpvOpFunctionParameter:
+          case SpvOpVariable:
+            // already processed
+            break;
+          case SpvOpLabel: {
+            // if previous instruction was early return, insert branch
+            // instruction
+            // to return block
+            if (prevInstWasReturn) {
+              if (returnLabelId == 0) returnLabelId = this->TakeNextId();
+              AddBranch(returnLabelId, &bp);
+              prevInstWasReturn = false;
+            }
+            // finish current block (if it exists) and get label for next block
+            uint32_t labelId;
```

Comment:
I do not see in the style guide that variables must be initialized. This variable is set on both paths of the following if. Still initialize? To zero?